src/vm/docs/mod.rs

```diff
@@ -536,6 +536,19 @@ If the function returns _ok_, database changes occurred.",
     example: "(contract-call! tokens transfer 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR 19) ;; Returns (ok 1)"
 };
 
+const AT_BLOCK: SpecialAPI = SpecialAPI {
+    input_type: "(buff 32), A",
+    output_type: "A",
+    signature: "(at-block block-hash expr)",
+    description: "The `at-block` function evaluates the expression `expr` _as if_ it were evaluated at the end of the
+block indicated by the _block-hash_ argument. The `expr` closure must be read-only.
+
+The function returns the result of evaluating `expr`.
+",
+    example: "(at-block 0x0000000000000000000000000000000000000000000000000000000000000000 (var-get data))"
```

Comment:
The thinking was that this is the most expressive version, since as in your example, users can use block-height via:

```
(at-block (get-block-info header-hash 100))
```

But also, if they chose to, specify an explicit block header.

Reply:
Adding that example in now.

Reply:
Thanks @kantai!